homeassistant/components/watchdog_file_watcher.py

```diff
@@ -0,0 +1,75 @@
+"""
+Component for monitoring activity on a folder.
+"""
+import asyncio
+import logging
+import voluptuous as vol
+from watchdog.observers import Observer
+from watchdog.events import PatternMatchingEventHandler
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+#DEPENDENCIES = ['watchdog']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PATH = 'folder'
+DOMAIN = "watchdog_file_watcher"
+EVENT_TYPE = "event_type"
+SRC_PATH = "src_path"
+PATTERNS = ["*.txt", "*.py", "*.md", "*.jpg", "*.png"]
```

Comment:
I don't think we should restrict file patterns like this, if at all.

Reply:
Make it configurable

Reply:
Agreed